src/main.rs

```diff
@@ -219,11 +271,17 @@ fn create_swarm(
 fn create_transport(
     identity: &identity::Keypair,
     websocket: bool,
+    tls: Option<tls::Config>,
 ) -> Result<Boxed<(PeerId, StreamMuxerBox)>> {
     let tcp_with_dns = TokioDnsConfig::system(TokioTcpConfig::new().nodelay(true)).unwrap();
 
     let transport = if websocket {
-        let websocket_with_dns = WsConfig::new(tcp_with_dns.clone());
+        let mut websocket_with_dns = WsConfig::new(tcp_with_dns.clone());
+
+        if let Some(tls) = tls {
+            websocket_with_dns.set_tls_config(tls);
+        }
```

Comment:
Side-note: We never dial so I _think_ we don't actually need DNS in here.

Reply:
good point, but I don't see it as part of this feature to remove (and test...) this.